### PR TITLE
rust toolchain version update

### PR DESCRIPTION
**Pull request by romoh**
**What this PR does / why we need it**:
Updates the toolchain version. This is required to ingest new updated dependencies that requires a more recent version of rustc.

### CODE REVIEW

**Review by bfjelds — APPROVED** [review]

File: discovery-handler-modules/debug-echo-discovery-handler/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "debug-echo-discovery-handler"
-version = "0.8.11"
+version = "0.8.12"
 authors = ["Kate Goldenring <kate.goldenring@microsoft.com>"]
 edition = "2018"
 rust-version = "1.58.1"
```
Comment (kate-goldenring) [inline discovery-handler-modules/debug-echo-discovery-handler/Cargo.toml:6]: The rust versions listed in all the cargo.tomls should be increased, as well.

**Review by kate-goldenring — CHANGES_REQUESTED** [review]
Thanks for pushing this up! Just a couple more spots we need the version increased: cargo.tomls and the badge in the base readme. After this merges, we need to also update the docs to point to htis latest version too: https://github.com/project-akri/akri-docs/pull/22

File: agent/src/util/device_plugin_service.rs [hunk 2]
```diff
@@ -746,8 +746,8 @@ pub async fn terminate_device_plugin_service(
 /// This creates an Instance's unique name
 pub fn get_device_instance_name(id: &str, config_name: &str) -> String {
     format!("{}-{}", config_name, &id)
-        .replace(".", "-")
-        .replace("/", "-")
+        .replace('.', "-")
+        .replace('/', "-")
```
[the remark this answers is not in the document]
Comment (romoh) [reply agent/src/util/device_plugin_service.rs:750]: yep, I changed it based on the clippy errors.

File: discovery-handler-modules/debug-echo-discovery-handler/Cargo.toml (hunk 1 above)
Comment (romoh) [reply discovery-handler-modules/debug-echo-discovery-handler/Cargo.toml:6]: Good catch

**Comment by kate-goldenring** [comment]
Looks like the tarpaulin version may need to be increased or parameterized to use the correct rust version: https://github.com/xd009642/tarpaulin

**Pull request merged**